modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -94,8 +99,13 @@ public ActionRequestValidationException validate() {
         if (destination.timestamp() != null) {
             e = addValidationError("setting timestamp on destination isn't supported. use scripts instead.", e);
         }
-        if (getRemoteInfo() != null && getSearchRequest().source().query() != null) {
-            e = addValidationError("reindex from remote sources should use RemoteInfo's query instead of source's query", e);
+        if (getRemoteInfo() != null) {
+            if (getSearchRequest().source().query() != null) {
+                e = addValidationError("reindex from remote sources should use RemoteInfo's query instead of source's query", e);
+            }
+            if (getWorkers() != 1) {
```

Comment:
I'd like to add workers support for reindex-from-remote in a followup. I don't think it'll be huge but I don't think I should make this PR even bigger than it already is.